content/blog/haskell-base-proposal.md

```diff
@@ -0,0 +1,99 @@
++++
+title = "Haskell base proposal"
+description = "A semi-concrete, semi-fleshed out proposal for changes to GHC and base"
+path = "/blog/2021/04/haskell-base-proposal"
+date = 2021-03-16
++++
+One of my greatest hopes for the Haskell Foundation is already beginning to be realized: better communication. I'm a member of the HF board and on the technical track. I've also been involved in some Discourse threads, and in particular [two](https://discourse.haskell.org/t/call-for-ideas-forming-a-technical-agenda/1901/33) [relevant](https://discourse.haskell.org/t/towards-a-prevalent-alternative-prelude/1986/74) threads for this blog post. I'm seeing a level of collaboration, interaction, and discussion of technical issues that has been lacking until now. I'm happy to see it happening, and hoping to see it continue.
+
+One of the most animated discussions has been around the topic of revising `base`, a new standard library, and alternative preludes. Plenty of my [previous](https://www.snoyman.com/blog/2020/10/haskell-bad-parts-1/) [blog](https://www.snoyman.com/blog/2020/11/haskell-bad-parts-2/) [posts](https://www.snoyman.com/blog/2020/12/haskell-bad-parts-3/) have already touched on my thoughts, but in a negative light: here are some problems. Today, I want to focus in on one area with a concrete solution. And I'm choosing this because I think it's an underappreciated problem, and one that we can actually solve, now that we have real collaboration between GHC, core library maintainers, and alternative prelude authors.
+
+## The problem
+
+If you want to represent a sequence of data in Haskell today, you have too many options:
+
+* A lazy list
+* A `Seq` (which, unfortunately, almost no one uses)
+* A strict and lazy `ByteString` for `Word8`s
+* A strict and lazy `Text` for `Char`s
+* A boxed vector (plus its mutable version)
+* An unboxed vector for types with `Unbox` instances
```

Comment:
You forgot to mention `primitive` package and its `ByteArray` and `Prim` instances. Without those `Unbox` class can't exist.

Reply:
I didn't really forget. I also "forgot" primitive vectors in the `vector` package. Both of those are marked as being not-for-end-users to some extent, so I'm leaving them out of the list.